### DISCUSSION TITLE
11RC1.1 IRenderTimer still available?

### DISCUSSION THREAD

**Discussion by msneijders** [AvaloniaUI/Avalonia#11624] [Q&A] [answered]
In a custom ScrollGestureRecognizer for inertia scrolling on touch devices I used `AvaloniaLocator.Current.GetService<Avalonia.Rendering.IRenderTimer>()` to get access to the render timer, to raise scroll events on each Tick. (The built-in ScrollGestureRecognizer uses a hardcoded 16ms timer to do this, but not all devices do 60fps).

Is it still posible for me to get to the RenderTimer?

**Comment by maxkatz6** [comment]
Why won't you use DispatcherTimer instead? Since you are raising Scroll events, it needs to be dispatched on the UI thread anyway.

**Reply by msneijders** [reply]
yes, I am dispatching it to ui from the tick -- but using DispatcherTimer you have to set an interval, that's why now its harcoded to 16ms (60fps), using rendertimer tick I can get it to the 'interval' of the renderer, so i can make the scroll work on the fps of the device (lke 120fps on my highend phone -- at least this seemed to work).

**Comment by kekekeks** [comment, marked as answer]
[We'll be adding](https://github.com/AvaloniaUI/Avalonia/issues/11627) `requestAnimationFrame`-like API before the release. 

Render timers are planned to be completely removed on some platforms in favor of simple vsync.

**Reply by msneijders** [reply]
FYI: just rewrote my custom ScrollGestureRecognizer using the new TopLevel.RequestAnimationFrame -- working as smooth as butter (and easier to use than rendertimer.tick :thumbsup:)
